api/cache.go

```diff
@@ -46,8 +46,10 @@ func (utc *userTierCache) Get(sub string) (int, bool) {
 	return ce.Tier, true
 }
 
-// Set stores the user's tier in the cache.
-func (utc *userTierCache) Set(u *database.User) {
+// Set stores the user's tier in the cache. If the customCacheKey is not empty,
+// it will be used to store the user in the cache, otherwise the user's sub will
+// be used.
+func (utc *userTierCache) Set(u *database.User, customCacheKey string) {
```

Comment:
This signature is a bit strange. I don't like the idea of being able to pass an optional value type and the caller being the one to figure out whether to pass `""` or not. So I'd probably just be explicit about it and make the second argument a `key` that's always either set to `u.Sub` or `akStr`.

I would also change the order of the arguments. Usually the key comes first followed by the value.